### PR TITLE
Thunks: Add support for disabling thunks in config

### PR DESCRIPTION
**Pull request by Sonicadvance1**
Previously the config options could only have ever enabled thunks rather than disable them.

Now sort the code so it can enable thunks, then following configs can redisable them.  Allowing testing with global thunks enabled and disabling problematic applications.

Also sorts the "ThunkConfigFile" config as lower priority than the application configs. I wasn't thinking about ordering that hard for these five configuration paths, but application configs should be higher priority in this case.

### CODE REVIEW

File: Source/Tests/LinuxSyscalls/FileManagement.cpp [hunk 1]
```diff
@@ -313,6 +278,44 @@ FileManager::FileManager(FEXCore::Context::Context *ctx)
     }
   }
 
+  // Now that we loaded the thunks object, walk through and ensure dependencies are enabled as well.
+  for (auto const &DBObject : ThunkDB) {
+    if (DBObject.second.Enabled) {
```
Comment (neobrain) [inline Source/Tests/LinuxSyscalls/FileManagement.cpp:283]: To keep indentation levels low:
```suggestion
    if (!DBObject.second.Enabled) {
        continue;
    }
```

File: Source/Tests/LinuxSyscalls/FileManagement.cpp [hunk 2]
```diff
@@ -313,6 +278,44 @@ FileManager::FileManager(FEXCore::Context::Context *ctx)
...
+      auto ThunkPath = ThunkGuestPath / DBObject.second.LibraryName;
+      if (std::filesystem::exists(ThunkPath)) {
+        for (auto Overlay : DBObject.second.Overlays) {
```
Comment (neobrain) [inline Source/Tests/LinuxSyscalls/FileManagement.cpp:286]: This avoids unnecessary copies of the contained `std::string`s:
```suggestion
        for (const auto& Overlay : DBObject.second.Overlays) {
```

File: Source/Tests/LinuxSyscalls/FileManagement.cpp [hunk 3]
```diff
@@ -313,6 +278,44 @@ FileManager::FileManager(FEXCore::Context::Context *ctx)
...
+      // Now walk the dependencies and set them up as well
+      // Make sure to enable each one as we go to remove circular dependencies
+      std::function<void(const std::unordered_set<std::string> &Depends)> InsertDependencies
+        = [this, &ThunkGuestPath, &InsertDependencies](const std::unordered_set<std::string> &Depends) -> void {
+        for (auto const &Depend : Depends) {
+          auto DBDepend = ThunkDB.find(Depend);
+          if (DBDepend != ThunkDB.end() &&
+              DBDepend->second.Enabled == false) {
+
+            auto ThunkPath = ThunkGuestPath / DBDepend->second.LibraryName;
```
Comment (neobrain) [inline Source/Tests/LinuxSyscalls/FileManagement.cpp:301]: This block is now duplicated outside the loop. Can we instead call `InsertDependencies({DBObject.first})` at the top level? That should make the loop body a tad more organized, and the extra lookup should be cheap.

**Review by neobrain — APPROVED** [review]
Added some minor comments, looks fine otherwise!

File: Source/Tests/LinuxSyscalls/FileManagement.cpp (hunk 1 above)
Comment (Sonicadvance1) [reply Source/Tests/LinuxSyscalls/FileManagement.cpp]: Resolved.

File: Source/Tests/LinuxSyscalls/FileManagement.cpp (hunk 2 above)
Comment (Sonicadvance1) [reply Source/Tests/LinuxSyscalls/FileManagement.cpp]: Resolved.

File: Source/Tests/LinuxSyscalls/FileManagement.cpp (hunk 3 above)
Comment (Sonicadvance1) [reply Source/Tests/LinuxSyscalls/FileManagement.cpp]: Resolved.

**Pull request merged**